locator/topology.cc

```diff
@@ -567,21 +567,27 @@ const endpoint_dc_rack& topology::get_location(const inet_address& ep) const {
 }
 
 void topology::sort_by_proximity(inet_address address, inet_address_vector_replica_set& addresses) const {
-    if (_sort_by_proximity) {
-        std::sort(addresses.begin(), addresses.end(), [this, &address](inet_address& a1, inet_address& a2) {
-            return compare_endpoints(address, a1, a2) < 0;
-        });
+    if (can_sort_by_proximity()) {
+        do_sort_by_proximity(address, addresses);
     }
 }
 
 void topology::sort_by_proximity(locator::host_id address, host_id_vector_replica_set& addresses) const {
```

Comment:
@gleb-cloudius do we still need duplicates of these functions?

Reply:
Eventually it will be remove. You just queued a series that removes some of its users. May be last once. Need to check.

Reply:
It looks like after the series to move streaming/repair to host id the function can be removed.

Reply:
There are still callers that use it with inet_address like: `range_streamer::get_all_ranges_with_sources_for`, `row_level_repair::sort_peer_nodes`, `storage_service::get_new_source_ranges`.

Cc @asias

Reply:
@bhalevy did you check in the next branch?

Reply:
No.  Now I see 5a849b0a6a29ff9ebcd748a45b69bed73da19473 (Merge "Move more subsystems to use host ids instead of ips" from Gleb)

Reply:
I'll rebase on top of it once it's merged to master